addons/base_vat/base_vat.py

```diff
@@ -328,3 +334,33 @@ def check_vat_tr(self, vat):
             return int(vat[9]) == c1 and int(vat[10]) == c2
 
         return False
+
+    def get_address(self, cr, uid, vat, context=None):
+        vals = {}
+        user_company = self.pool['res.users'].browse(cr, uid, uid, context=context).company_id
+        if user_company.vat_check_vies:
+            try:
+                vat_country, vat_number = self._split_vat(vat)
+                client = Client(vatnumber.VIES_URL)
+                response = client.service.checkVat(countryCode=vat_country, vatNumber=vat_number)
+                if response['valid']:
+                    country_id = self.pool['res.country'].search(cr, uid, [('code', '=ilike', vat_country)], context=context)
+                    address = str(response['address'].encode('utf-8')).split('\n')
+                    vals.update(street=address[0], street2=address[1] if len(address) > 1 else False, country_id=country_id[0] if country_id else False)
+            except:
+                # see http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl
+                # Fault code may contain INVALID_INPUT, SERVICE_UNAVAILABLE, MS_UNAVAILABLE,
```

Comment:
Too many fault code unmanaged. We should catch all the error code and give a proper warning